racket/collects/racket/contract/private/misc.rkt

```diff
@@ -706,19 +696,20 @@
    (none/c-name ctc)
    val))
 
-(define-struct none/c (name)
+(struct none/c (name)
+  #:constructor-name make-none/c
   #:property prop:custom-write custom-write-property-proc
-  #:omit-define-syntaxes
+  #:property prop:none/c #f
   #:property prop:flat-contract
   (build-flat-contract-property
    #:trusted trust-me
    #:late-neg-projection none-curried-late-neg-proj
-   #:stronger (λ (this that) #t)
-   #:equivalent (λ (this that) (none/c? that))
+   #:stronger (λ (this that) (prop:none/c? that))
```

Comment:
I think this change is incorrect. If there aren't test cases that this change breaks, let's add some.